### PR TITLE
Removed init lib in favor of settings lib

### PR DESCRIPTION
**Pull request by pwnsdx**
- Less usage of disk
- Only one read of the config file is required
- Only one async write required
- Cleaner class

### CODE REVIEW

File: electron/js/lib/ConfigurationPersistence.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+'use strict';
+
+const fs = require('fs');
+const path = require('path');
+
+const app = require('electron').app || require('electron').remote.app;
```
Comment (bennyn) [inline electron/js/lib/ConfigurationPersistence.js]: Do we need the `||`?

File: electron/js/lib/ConfigurationPersistence.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    const value = global._ConfigurationPersistence[name];
+    return (typeof value !== 'undefined' ? value : defaultValue);
+  }
+
+  _saveToFile() {
+    return new Promise((resolve, reject) => {
+      const datasInJSON = JSON.stringify(global._ConfigurationPersistence);
+      this.debug('Saving datas to persistent storage: %o', datasInJSON);
+      fs.writeFile(INIT_JSON, datasInJSON, 'utf8', (err, data) => {
+        if (err) {
```
Comment (bennyn) [inline electron/js/lib/ConfigurationPersistence.js]: Sometimes it's called `err`, sometimes it's called `e` and sometimes `error`. Let's stick to the most explicit one which is `error`.

File: electron/js/lib/ConfigurationPersistence.js (hunk 1 above)
Comment (pwnsdx) [inline electron/js/lib/ConfigurationPersistence.js]: No idea but it was here originally so

File: electron/main.js [hunk 3]
```diff
@@ -323,15 +323,23 @@ function showMainWindow() {
...
     if (!main.isFullScreen()) {
       init.save('bounds', main.getBounds());
     }
 
     if (!quitting) {
       event.preventDefault();
       main.hide();
+
+    } else {
+
```
Comment (bennyn) [inline electron/main.js]: Why there is an empty line?

File: electron/js/lib/ConfigurationPersistence.js [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+const app = require('electron').app || require('electron').remote.app;
+const INIT_JSON = path.join(app.getPath('userData'), 'init.json');
+const debug = require('debug');
+
+class ConfigurationPersistence {
+
+  constructor() {
+    this.debug = debug('ConfigurationPersistence');
+
+    // Get content of init.json
```
Comment (bennyn) [inline electron/js/lib/ConfigurationPersistence.js]: No need for having this comment. It's clear by the code.

File: electron/main.js [hunk 5]
```diff
@@ -323,15 +323,23 @@ function showMainWindow() {
...
+      // Save configuration to disk
```
Comment (lipis) [inline electron/main.js]: no need for this comment.. pretty clear

File: electron/js/menu/system.js [hunk 6]
```diff
@@ -24,9 +24,9 @@ const autoLaunch = require('auto-launch');
 const launchCmd = (process.env.APPIMAGE != null) ? process.env.APPIMAGE : process.execPath;
 
 const config = require('./../config');
-const init = require('./../lib/init');
 const locale = require('./../../locale/locale');
 const windowManager = require('./../window-manager');
+const init = require('./../lib/ConfigurationPersistence');
```
Comment (lipis) [inline electron/js/menu/system.js]: they were sorted alphabetically.. so let's keep it that way

Comment (pwnsdx) [inline electron/js/menu/system.js]: Already fixed

**Comment by lipis** [comment]
init was a bit general.. maybe we could name it `settings` instead of ConfigurationPersistence?

File: electron/js/lib/ConfigurationPersistence.js (hunk 1 above)
Comment (pwnsdx) [inline electron/js/lib/ConfigurationPersistence.js]: Apparently used so it can be compatible with both main process and renderer

Comment (lipis) [inline electron/js/lib/ConfigurationPersistence.js]: yes we need to keep it

**Pull request merged**